lib/spark/options/options.ex

```diff
@@ -1229,6 +1232,34 @@ defmodule Spark.Options do
       error_tuple(key, value, "expected `#{inspect(protocol)}` to be a protocol")
   end
 
+  defp validate_type({:impl, protocol}, key, value) do
+    protocol.__info__(:module)
```

Comment:
I see that we're doing `protocol.impl_for!` in our other example, but I'm actually not sure this is a great idea as it hard-locks a compile time dependency on the module that you place in the option. We can start with it here, but we may need to either modify this or add something like `:impl_no_verify` and `:protocol_no_verify` later on to avoid. this.

Reply:
I am not sure why that can be an issue. The validation is performed only when the DSL is actually used, and if the option is required, the dependency is already implied. If it is optional, the check will only be performed when the option is actually used

Reply:
In many cases we intentionally avoid compile time dependencies when working with DSLs. It's more common in the case of behaviors than protocols. In spark you can do that with

```elixir
schema: [
  thing: {:behaviour, ...}
],
modules: [:thing]
```

The behaviour type does not validate for that reason. Unnecessary compilation dependency avoidance is a very important aspect of Spark. If you were to do this work in a function, then there would only be a runtime dependency, but with spark, it incurs a compile time dependency on the target module. This can be very hard to untangle in some cases.